driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -211,17 +210,48 @@ public interface Expression {
      * the user is an unchecked assertion that all elements are of that type,
      * and that no element is null, is missing, or is of some other type. If the
      * user cannot make such an assertion, some appropriate super-type should be
-     * chosen, and if necessary elements should be individually type-checked.
+     * chosen, and if necessary the elements should be individually type-checked.
      *
      * @param other the other value.
      * @return the resulting value.
      * @param <T> the type of the elements of the resulting array.
      */
     <T extends Expression> ArrayExpression<T> isArrayOr(ArrayExpression<? extends T> other);
 
-    // TODO-END doc after Map merged, "record" and "schema objects" are decided
+    /**
+     * {@code this} value as a {@linkplain DocumentExpression document} if
+     * {@code this} is a document or document-like value (such as a
```

Comment:
I have updated the wording.

I think it is better to avoid "users must rely" phrasing since the implication/obligation seems much broader than simply specifying method behaviour.

When it comes to explaining document/map/entry, it seems better to plainly state that we offer "multiple abstractions of certain types (document, map, entry)".

Reply:
[optional]

> all {@linkplain EntryExpression entries} created using the present API

If we introduce ways to create non-document-like entries in the future, the phrase "present API" will automatically start wrongly encompassing them. If we instead list all the ways that produce document-like entires, then we save ourselves from such a problem.

Zooming out, I don't see a reason you want users to consider entries document-like, given that the structure of such a document is not specified.